### PR TITLE
Schema entites

### PR DESCRIPTION
**Pull request by warfawl**

### CODE REVIEW

File: src/main/java/ua/danit/final_project/entities/Task.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  @Column(name = "updated", nullable = false)
+  private Timestamp updated;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(
+          name = "task_location",
+          joinColumns = {@JoinColumn(name = "t_id")},
+          inverseJoinColumns = {@JoinColumn(name = "l_id")})
+  List<Location> locations;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/Task.java]: Code Smell Code Smell: Make "locations" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZfD2xJdnw6Qj_eV&open=AWVYzZfD2xJdnw6Qj_eV)

[ISSUE_KEY]: AWVYzZfD2xJdnw6Qj_eV

File: src/main/java/ua/danit/final_project/entities/Location.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Column(name = "l_info")
+  private String info;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(
+          name = "task_location",
+          joinColumns = {@JoinColumn(name = "l_id")},
+          inverseJoinColumns = {@JoinColumn(name = "t_id")})
+  List<Task> tasks;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/Location.java]: Code Smell Code Smell: Make "tasks" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZew2xJdnw6Qj_eR&open=AWVYzZew2xJdnw6Qj_eR)

[ISSUE_KEY]: AWVYzZew2xJdnw6Qj_eR

File: src/main/java/ua/danit/final_project/entities/Permission.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Column(name = "p_name", nullable = false, unique = true)
+  private String name;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(
+          name = "role_permission",
+          joinColumns = {@JoinColumn(name = "p_id")},
+          inverseJoinColumns = {@JoinColumn(name = "r_id")})
+  public List<Role> roles;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/Permission.java]: Code Smell Code Smell: Make "roles" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZe12xJdnw6Qj_eS&open=AWVYzZe12xJdnw6Qj_eS)

[ISSUE_KEY]: AWVYzZe12xJdnw6Qj_eS

File: src/main/java/ua/danit/final_project/entities/Role.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          name = "role_permission",
+          joinColumns = {@JoinColumn(name = "r_id")},
+          inverseJoinColumns = {@JoinColumn(name = "p_id")})
+  public List<Permission> permissions;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(name = "user_role",
+          joinColumns = {@JoinColumn(name = "r_id")},
+          inverseJoinColumns = {@JoinColumn(name = "u_id")})
+  public List<User> users;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/Role.java]: Code Smell Code Smell: Make "users" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZe82xJdnw6Qj_eU&open=AWVYzZe82xJdnw6Qj_eU)

[ISSUE_KEY]: AWVYzZe82xJdnw6Qj_eU

File: src/main/java/ua/danit/final_project/entities/User.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  @JsonIgnore
+  @Column(name = "u_password", nullable = false)
+  private String password;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(name = "user_role",
+          joinColumns = {@JoinColumn(name = "u_id")},
+          inverseJoinColumns = {@JoinColumn(name = "r_id")})
+  public List<Role> roles;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/User.java]: Code Smell Code Smell: Make "roles" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZfK2xJdnw6Qj_eW&open=AWVYzZfK2xJdnw6Qj_eW)

[ISSUE_KEY]: AWVYzZfK2xJdnw6Qj_eW

File: src/main/java/ua/danit/final_project/entities/Role.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  @Column(name = "r_name", unique = true, nullable = false)
+  private String name;
+
+  @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinTable(
+          name = "role_permission",
+          joinColumns = {@JoinColumn(name = "r_id")},
+          inverseJoinColumns = {@JoinColumn(name = "p_id")})
+  public List<Permission> permissions;
```
Comment (crystweb) [inline src/main/java/ua/danit/final_project/entities/Role.java]: Code Smell Code Smell: Make "permissions" private or transient. ([squid:S1948](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1948))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=82&issues=AWVYzZe82xJdnw6Qj_eT&open=AWVYzZe82xJdnw6Qj_eT)

[ISSUE_KEY]: AWVYzZe82xJdnw6Qj_eT

**Comment by warfawl** [comment]
[Define list of all Entities in the application, define list of minimum required properties for each entity](https://trello.com/c/VwkfaLCw/49-define-list-of-all-entities-in-the-application-define-list-of-minimum-required-properties-for-each-entity)

**Comment by warfawl** [comment]
[Add all Entities to the application](https://trello.com/c/zgR2Xy86/51-add-all-entities-to-the-application)

**Pull request merged**